### ISSUE TITLE
Fix request: false positive in HttpClientBuilderWithoutTimeouts 

### ISSUE THREAD

**Issue by narasimha3841** [jborgers/PMD-jPinpoint-rules#163]
I had to configure timeouts two times for the below code in my rest template request factory.  I got sonarQube violations on HttpComponentsClientHttpRequestFactory & HttpClientBuilder 

    private HttpComponentsClientHttpRequestFactory getRequestFactory(SSLContext sslContext) {
        RequestConfig requestConfig = RequestConfig.custom()
            .setConnectionRequestTimeout(properties.getConnectionRequestTimeOutInMillis())
            .setConnectTimeout(properties.getConnectTimeOutInMillis())
            .setSocketTimeout(properties.getReadTimeOutInMillis())
            .build();

        HttpComponentsClientHttpRequestFactory clientHttpRequestFactory = new HttpComponentsClientHttpRequestFactory(
                HttpClients.custom()
                        .setDefaultRequestConfig(requestConfig)
                        .setSSLContext(sslContext)
                        .setSSLSocketFactory(
                                new SSLConnectionSocketFactory(
                                        sslContext,
                                        SUPPORTED_PROTOCOLS.toArray(new String[0]),
                                        null,
                                        SSLConnectionSocketFactory.getDefaultHostnameVerifier()
                                )
                        )
                        .setMaxConnTotal(DEFAULT_MAXIMUM_CONNECTIONS_TOTAL)
                        .setMaxConnPerRoute(DEFAULT_MAXIMUM_CONNECTIONS_PER_ROUTE)
                        .disableConnectionState()
                        .setDefaultHeaders(initHeaders())
                        .build()
        );

        clientHttpRequestFactory.setConnectTimeout(properties.getConnectTimeOutInMillis());
        clientHttpRequestFactory.setConnectionRequestTimeout(properties.getConnectionRequestTimeOutInMillis());
        clientHttpRequestFactory.setReadTimeout(properties.getReadTimeOutInMillis());

        return clientHttpRequestFactory;
    }

**Comment by BertKoor** [comment]
Got a similar issue. Refactored the code a while ago to let pinpoint-rule understand all timeouts were set, now something in pinpoint-rule has changed and seeing it as a CRITICAL issue again.

````
    public static HttpComponentsClientHttpRequestFactory createHttpRequestFactory(SSLContext sslContext, ConnectionProperties properties) {
        CloseableHttpClient client = HttpClients.custom()
// pmd:HttpClientBuilderWithoutTimeouts: HttpClient builder is used and not all three timeouts are configured. The defaults are probably not optimal (e.g. infinite).
                .setSSLContext(sslContext)
                .setMaxConnTotal(properties.getMaxTotalConnections())
                .setMaxConnPerRoute(properties.getMaxConnectionsPerRoute())
                .disableConnectionState().build();

        return createAndConfigureHttpRequestFactory(client, properties);
    }

    private static HttpComponentsClientHttpRequestFactory createAndConfigureHttpRequestFactory(CloseableHttpClient client, ConnectionProperties properties) {
        HttpComponentsClientHttpRequestFactory factory = new HttpComponentsClientHttpRequestFactory(client);
        factory.setConnectionRequestTimeout(properties.getRequestTimeoutMillis());
        factory.setConnectTimeout(properties.getConnectTimeoutMillis());
        factory.setReadTimeout(properties.getReadTimeoutMillis());
        return factory;
    }

````

**Comment by jborgers** [comment]
@BertKoor Could you run the current jpinpoint-rules on your code? I cannot reproduce your violation. Maybe the Sonar ruleset is behind on this issue.

**Comment by jborgers** [comment]
@narasimha3841 So you want to be able to set the timeouts only on the RequestConfig right?

**Comment by BertKoor** [comment]
> jpinpoint

Did rerun, SonarQube at rabobank still flags this piece of code as critical violation, but I cannot determine against what version (or fork) of the rules this is checked against. Kinda gave up, long time ago we put in 
@SuppresWarnings("pmd:HttpClientBuilderWithoutTimeouts") // false positive
